File: cpp/leetcode/DataStreamIntervals.cpp
//Leetcode Problem No 352 Data Stream as Disjoint Intervals
//Solution written by Xuqiang Fang on 5 June, 2018

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <unordered_map>
#include <unordered_set>
#include <stack>
#include <queue>

using namespace std;
struct Interval {
    int start;
    int end;
    Interval() : start(0), end(0) {}
    Interval(int s, int e) : start(s), end(e) {}
};
class SummaryRanges {
public:
    /** Initialize your data structure here. */
    SummaryRanges() {
        
    }
    
    void addNum(int val) {
         
    }
    
    vector<Interval> getIntervals() {
        
    }
private:
    unordered_map<int, Interval> map_;
};

int main(){
    SummaryRanges s;

    return 0;
}

File: java/leetcode/DataStreamIntervals.java
//Leetcode problem 352 Data Stream as Disjoint Intervals
//Solution written by Xuqiang Fang on 5 June, 2018
import java.util.ArrayList;
import java.util.List;
import java.util.HashMap;
import java.util.Map;
import java.util.HashSet;
import java.util.Set;
import java.util.PriorityQueue;
import java.util.Stack;
import java.util.Queue;
import java.util.LinkedList;
import java.util.Arrays;

class Interval {
    int start;
    int end;
    Interval() { start = 0; end = 0; }
    Interval(int s, int e) { start = s; end = e; }
}
class SummaryRanges {
    /** Initialize your data structure here. */
    public SummaryRanges() {
        map = new TreeMap<>(); 
    }
    
    public void addNum(int val) {
         
    }
    
    public List<Interval> getIntervals() {
        
    }
    private TreeMap<Integer, Interval> map;
}

public class {
	public static void main(String[] args){
		Solution s = new Solution();
	}
}
